gradle-sls-packaging/src/test/groovy/com/palantir/gradle/dist/tasks/CreateManifestTaskIntegrationSpec.groovy

```diff
@@ -332,10 +334,110 @@ class CreateManifestTaskIntegrationSpec extends GradleIntegrationSpec {
         """.stripIndent())
 
         when:
-        runTasks(':foo-server:createManifest', '-i')
+        runTasks(':foo-server:createManifest', '-i', '--write-locks')
+
+        then: 'foo-server does not include transitively discovered self dependency'
+        !fileExists('foo-server/product-dependencies.lock')
+    }
+
+    def 'masks minimum version in product dependency that is published by this repo if same as project version'() {
+        setup:
+        buildFile << """
+        allprojects {
+            project.version = '1.0.0-rc1.dirty'
+        }
+        """
+        helper.addSubproject("foo-api", """
+            apply plugin: 'java'
+            apply plugin: 'com.palantir.sls-recommended-dependencies'
+            
+            recommendedProductDependencies {
+                productDependency {
+                    productGroup = 'com.palantir.group'
+                    productName = 'foo-service'
+                    minimumVersion = rootProject.version
+                    maximumVersion = '1.x.x'
+                    recommendedVersion = rootProject.version
+                }
+            }
+        """.stripIndent())
+        helper.addSubproject("foo-server", """
+            apply plugin: 'com.palantir.sls-java-service-distribution'
+            distribution {
+                serviceGroup 'com.palantir.group'
+                serviceName 'foo-service'
+                mainClass 'com.palantir.foo.bar.MyServiceMainClass'
+                args 'server', 'var/conf/my-service.yml'
+            }
+        """.stripIndent())
+        helper.addSubproject("bar-server", """
+            apply plugin: 'com.palantir.sls-java-service-distribution'
+            dependencies {
+                compile project(':foo-api')
+            }
+            distribution {
+                serviceGroup 'com.palantir.group'
+                serviceName 'bar-service'
+                mainClass 'com.palantir.foo.bar.MyServiceMainClass'
+                args 'server', 'var/conf/my-service.yml'
+            }
+        """.stripIndent())
+
+        when:
+        runTasks('--write-locks')
+
+        then:
+        file('bar-server/product-dependencies.lock').readLines().contains 'com.palantir.group:foo-service (project-version, 1.x.x)'
```

Comment:
Could we use `$project-version` instead of just `project-version` to emphasise this value is somehow special?